core/store/hashtree/binprefix/tree.go

```diff
@@ -308,23 +315,39 @@ type EmptyNode struct {
 
 // NewEmptyNode creates a new empty node.
 func NewEmptyNode(depth uint16, key *big.Int) *EmptyNode {
+	return NewEmptyNodeWithDigest(depth, key, nil)
+}
+
+// NewEmptyNodeWithDigest creates a new empty node with its digest.
+func NewEmptyNodeWithDigest(depth uint16, key *big.Int, hash []byte) *EmptyNode {
 	return &EmptyNode{
 		depth:  depth,
 		prefix: key,
+		hash:   hash,
 	}
 }
 
-// GetHash implements mem.TreeNode. It returns the hash of the node.
+// GetHash implements binprefix.TreeNode. It returns the hash of the node.
 func (n *EmptyNode) GetHash() []byte {
 	return append([]byte{}, n.hash...)
 }
 
-// GetType implements mem.TreeNode. It returns the empty node type.
+// GetType implements binprefix.TreeNode. It returns the empty node type.
 func (n *EmptyNode) GetType() byte {
 	return emptyNodeType
 }
 
-// Search implements mem.TreeNode. It always return a empty value.
+// GetDepth returns the depth of the node.
+func (n *EmptyNode) GetDepth() uint16 {
+	return n.depth
+}
+
+// GetPrefix returns the prefix of the node.
+func (n *EmptyNode) GetPrefix() *big.Int {
+	return n.prefix
+}
+
+// Search implements binprefix.TreeNode. It always return a empty value.
```

Comment:
those functions often take the bucket in argument but don't use it. Is it to match some existing APIs ?

Reply:
Yep it's mostly to load disk nodes but it is required in the tree node interface.